frontend/src/employee-frontend/components/income-statements/IncomeStatementsPage.tsx

```diff
@@ -58,20 +63,67 @@ function IncomeStatementsList({
 export default React.memo(function IncomeStatementsPage() {
   const { i18n } = useTranslation()
 
-  const [result, setResult] = useState<
-    Result<IncomeStatementAwaitingHandler[]>
+  const [areas, setAreas] = useState<Result<CareArea[]>>(Loading.of())
+
+  const [incomeStatements, setIncomeStatements] = useState<
+    Result<Paged<IncomeStatementAwaitingHandler> & { currentPage: number }>
   >(Loading.of())
-  const loadData = useRestApi(getIncomeStatementsAwaitingHandler, setResult)
-  useEffect(() => loadData(), [loadData])
+  const [toggledAreas, setToggledAreas] = useState<string[]>([])
+
+  const loadAreas = useRestApi(getAreas, setAreas)
+  const loadData = useRestApi(
+    getIncomeStatementsAwaitingHandler,
+    setIncomeStatements
+  )
+
+  useEffect(() => {
+    loadAreas()
+    loadData()
+  }, [loadAreas, loadData])
+
+  useEffect(() => {
+    loadData(toggledAreas)
+  }, [loadData, toggledAreas])
+
+  const toggleArea = (area: string) => () => {
+    setToggledAreas((prev) => {
+      if (prev.includes(area)) return prev.filter((t) => t !== area)
+      else return [...prev, area]
+    })
+  }
+
+  const handlePageChange = useCallback(
+    (page: number) => {
+      loadData(toggledAreas, page)
```

Comment:
Instead of triggering the `loadData` manually from multiple places, I think it would be slightly more concise to just update the fetch parameters and have the loadData rely on param changes.

Reply:
Even better catch! Fixed.